File: sol/3461/3461.cpp
struct Solution {
    bool hasSameDigits(string &s) {
        int n = s.size();
        for (char &c : s){
            c -= '0';
        }
        while (n-- > 2) {
            for (int i = 0; i < n; ++i){
                s[i] = (s[i] + s[i + 1]) % 10;
            }
        }
        return s[0] == s[1];
    }
};

File: sol/3461/3461.py
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        tpl = list(map(int, s))
        while len(tpl) != 2:
            tpl = [(a + b) % 10 for a, b in zip(tpl[:-1], tpl[1:])]
        return tpl[0] == tpl[1]
